src/borg/locking.py

```diff
@@ -158,6 +161,41 @@ def is_locked(self):
     def by_me(self):
         return os.path.exists(self.unique_name)
 
+    def kill_dead_lock(self):
+        for name in os.listdir(self.path):
+
+            try:
+                host_pid, thread = name.rsplit('-', 1)
+                host, pid = host_pid.rsplit('.', 1)
+            except ValueError:
+                # Malformed lock name? Or just soem new format we don't understand?
+                # It's safer to just exit
+                return False
+
+            if host != _hostname:
+                return False
+
+            try:
+                # This may not work in Windows.
+                os.kill(int(pid), 0)
+                return False
+            except OSError:
+                pass
```

Comment:
one such circumstance that I was after is that the proess does not exist.

I guess another somewhat feasible one is "permission denied". We probably need to catch it separately, even though we'll fail with exception trying to delete the lock too (do we want to catch that anyway?)
In general I assume it's a bad idea running borg instances from different users on the local box.